packages/cli/generation/remote-generation/remote-workspace-runner/src/publishDocs.ts

```diff
@@ -218,9 +218,79 @@ async function convertDocsConfiguration({
         },
         colors: docsDefinition.config.colors,
         navbarLinks: docsDefinition.config.navbarLinks,
+        typography: convertDocsTypographyConfiguration({
+            typographyConfiguration: docsDefinition.config.typography,
+            docsDefinition,
+            uploadUrls,
+            context,
+        }),
     };
 }
 
+function convertDocsTypographyConfiguration({
+    typographyConfiguration,
+    docsDefinition,
+    uploadUrls,
+    context,
+}: {
+    typographyConfiguration?: TypographyConfig;
+    docsDefinition: DocsDefinition;
+    uploadUrls: Record<FernRegistry.docs.v1.write.FilePath, FernRegistry.docs.v1.write.FileS3UploadUrl>;
+    context: TaskContext;
+}): FernRegistry.docs.v1.write.DocsTypographyConfig | undefined {
+    if (typographyConfiguration == null) {
+        return;
+    }
+    const result: FernRegistry.docs.v1.write.DocsTypographyConfig = {};
+
+    if (typographyConfiguration.headingsFont != null) {
```

Comment:
Not a huge deal but I could see us simplifying this function to just something like:

```ts
    if (typographyConfiguration == null) {
        return undefined;
    }
    return {
        headingsFont: convertFont({
            font: typographyConfiguration.headingsFont,
            docsDefinition,
            context,
            uploadUrls,
            label: "heading"
        }),
        bodyFont: convertFont({
            font: typographyConfiguration.bodyFont,
            docsDefinition,
            context,
            uploadUrls,
            label: "body"
        }),
        codeFont: convertFont({
            font: typographyConfiguration.codeFont,
            docsDefinition,
            context,
            uploadUrls,
            label: "code"
        }),
    }
```

where convert font is something like

```ts
function convertFont(...) {
    if (font == null) {
        return undefined
    }

    const filepath = convertAbsoluteFilepathToFdrFilepath(
        font.absolutePath,
        docsDefinition
    );
    const file = uploadUrls[filepath];
    if (file == null) {
        return context.failAndThrow(`Failed to locate ${label} file after uploading`);
    }
    result.headingsFont = {
        name: font.name ?? `font:${label}:${file.fileId}`,
        fontFile: file.fileId,
    };    
}
```

Reply:
Thanks! This was definitely a lot cleaner. Added changes